llvmlite/ir/types.py

```diff
@@ -122,11 +122,13 @@ class PointerType(Type):
 
     # Factory to create typed or opaque pointers based on `pointee'.
     def __new__(cls, pointee=None, addrspace=0):
-        if cls is PointerType and pointee is not None:
+        if cls is PointerType and pointee is not None and \
+           type(pointee) is not PointerType:
             return super().__new__(_TypedPointerType)
         return super(PointerType, cls).__new__(cls)
 
-    def __init__(self, addrspace=0):
+    def __init__(self, __pointee=None, addrspace=0):
```

Comment:
```suggestion
    def __init__(self, pointee=None, /, addrspace=0):
```

The use of `__` prefix in argument seems unusual. 
```python
>>> class A:
...     def __init__(self, __p=0):
...         self.p = __p
...
>>> A(__p=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: A.__init__() got an unexpected keyword argument '__p'

>>> import inspect
>>> inspect.signature(A.__init__)
<Signature (self, _A__p=0)>
```
If we want it to be positional-only, we should add `/` before `addrspace`.

Reply:
Thanks, it should be fixed now. :)